pipeline/dags/dag_utils/marts.py

```diff
@@ -1,44 +1,33 @@
-from airflow.operators import bash, python
+from airflow.operators import python
 
 from dag_utils.virtualenvs import PYTHON_BIN_PATH
 
-TMP_FILE_PATH = "/tmp/out.dump"
-
-
-def _export_to_s3(tmp_file_path, logical_date, run_id):
-    import logging
 
+def _export_di_dataset_to_s3(logical_date, run_id):
     from airflow.providers.amazon.aws.hooks import s3
+    from airflow.providers.postgres.hooks import postgres
 
     from dag_utils import date
 
-    logger = logging.getLogger(__name__)
-    s3_file_path = f"data/marts/{date.local_date_str(logical_date)}/{run_id}/api.dump"
+    pg_hook = postgres.PostgresHook(postgres_conn_id="pg")
+    s3_hook = s3.S3Hook(aws_conn_id="s3")
+
+    prefix = f"data/marts/{date.local_date_str(logical_date)}/{run_id}/"
 
-    logger.info("Uploading dump to s3_path=%s", s3_file_path)
+    for ressource in ["structures", "services"]:
+        # TODO(vmttn): normalize table name after db split
+        table_name = ressource.rstrip("s")
+        key = f"{prefix}{ressource}.parquet"
+        query = f"SELECT * FROM public.{table_name}"
 
-    s3_hook = s3.S3Hook(aws_conn_id="s3")
-    s3_hook.load_file(filename=tmp_file_path, key=s3_file_path)
-
-
-def pg_dump_api():
-    return bash.BashOperator(
-        task_id="pg_dump_api",
-        bash_command=(
-            "pg_dump --format=custom "
-            "--dbname={{ conn.pg.get_uri() }} "
-            "--schema=public "
-            "--table=service "
-            "--table=structure "
-            f"--file={TMP_FILE_PATH}"
-        ),
-    )
+        df = pg_hook.get_pandas_df(sql=query)
+        bytes_data = df.to_parquet(compression="gzip")
+        s3_hook.load_bytes(bytes_data, key=key, replace=True)
```

Comment:
Je trouve qu'on devrait mettre plus de logs. On en a déjà fort peu et c'est toujours penible d'investiguer après coup.

Là, on a concentré deux étapes en une et ajouté un `for` qui va rendre encore plus difficile le fait de savoir sur quelle table et à quelle étape ça a planté.
Peut-être mettre quelques marqueurs ? (afficher un peek de la dataframe + le nom de la table, rappeler la requete SQL complète et le nom du fichier final dans S3 entre les étapes qui peuvent échouer, par exemple)

Reply:
donc les `for` sont à proscrire ? non sérieusement tu as raison c'est mieux avec les marqueurs, mais n'hésites pas à t'en occuper, je ne crois pas qu'ils étaient tous bien présent dans ton travail initial

Reply:
Les for ne sont pas à proscrire, mais ils ne simplifient pas le débug sans log :)

Et en effet j'en avais mis un seul timide au départ mais ça allait un poil mieux avec un log + 2 tâches distinctes. Cela dit cette version est top, je vais juste mettre un petit log par ci par là.